desc/compute/_bootstrap.py

```diff
@@ -0,0 +1,504 @@
+        import matplotlib.pyplot as plt
+
+        plt.figure(figsize=(14, 7))
+        plt.rcParams.update({"font.size": 8})
+        nrows = 5
+        ncols = 5
+        variables = [
+            "Bmax",
+            "Bmin",
+            "epsilon",
+            "<B**2>",
+            "<1/B>",
+            "f_t",
+            "iota",
+            "G",
+            "R",
+            "ne_rho",
+            "ni_rho",
+            "Zeff_rho",
+            "Te_rho",
+            "Ti_rho",
+            "ln_Lambda_e",
+            "ln_Lambda_ii",
+            "nu_e_star",
+            "nu_i_star",
+            "dnds_term",
+            "dTeds_term",
+            "dTids_term",
+            "L31",
+            "L32",
+            "alpha",
+            "<J*B>",
+        ]
+        for j, variable in enumerate(variables):
+            plt.subplot(nrows, ncols, j + 1)
+            plt.plot(rho, J_dot_B_data[variable])
+            plt.title(variable)
+            plt.xlabel(r"$\rho$")
+        plt.tight_layout()
+        plt.show()
+
+    return J_dot_B_data
+
+
+@register_compute_fun(
+    name="<J*B> Redl",
+    label="\\langle\\mathbf{J}\\cdot\\mathbf{B}\\rangle_{Redl}",
+    units="T A m^{-2}",
+    units_long="Tesla Ampere / meter^2",
+    description="Bootstrap current profile, Redl model for quasisymmetry",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[
+        "electron_density",
+        "electron_temperature",
+        "ion_temperature",
+        "atomic_number",
+    ],
+    coordinates="r",
+    data=["|B|", "sqrt(g)", "G", "I", "iota"],
+)
+def _compute_J_dot_B_Redl(params, transforms, profiles, data, **kwargs):
+    r"""Compute the bootstrap current.
+
+    (specifically :math:`\left<\vec{J}\cdot\vec{B}\right>`) using the formulae in
+    Redl et al, Physics of Plasmas 28, 022502 (2021). This formula for
+    the bootstrap current is valid in axisymmetry, quasi-axisymmetry,
+    and quasi-helical symmetry, but not in other stellarators.
+    """
+    grid = transforms["grid"]
+
+    # Note that the geom_data dictionary provided to j_dot_B_Redl()
+    # contains info only as a function of rho, not theta or zeta,
+    # i.e. on the compressed grid. In contrast, "data" contains
+    # quantities on a 3D grid even for quantities that are flux
+    # functions.
+    geom_data = trapped_fraction(grid, data["|B|"], data["sqrt(g)"])
+    geom_data["G"] = compress(grid, data["G"])
+    geom_data["I"] = compress(grid, data["I"])
+    geom_data["iota"] = compress(grid, data["iota"])
+    geom_data["R"] = (geom_data["G"] + geom_data["iota"] * geom_data["I"]) * geom_data[
+        "<1/B>"
+    ]
+    geom_data["psi_edge"] = params["Psi"] / (2 * jnp.pi)
+
+    ne = profiles["electron_density"]
+    Te = profiles["electron_temperature"]
+    Ti = profiles["ion_temperature"]
+    Zeff = profiles["atomic_number"]
+    # The "backup" PowerSeriesProfiles that follow here are necessary
+    # for test_compute_funs.py::test_compute_everything
```

Comment:
I think this may be a non-issue now that ne/Te/Ti/Zeff are obtained using data dependencies - `test_compute_everything` seems to work fine for the bootstrap compute function.